js/main.js

```diff
@@ -1,2 +1,29 @@
 /* global data */
 /* exported data */
+function imgUrlInput(event) {
+  var imgsrc = event.target.value;
+  $imageDisplay.setAttribute('src', imgsrc);
+}
+function formJournalSubmit(event) {
+  var newObj = {};
+  event.preventDefault();
+  newObj.title = $formJournal.elements.title.value;
+  newObj.imgsrc = $formJournal.elements.photoUrl.value;
+  newObj.notes = $formJournal.elements.notes.value;
+  newObj.nextEntryId = data.nextEntryId;
+  data.entryId = (data.nextEntryId) + 1;
+  data.entries.push(newObj);
+  var jsonEntries = JSON.stringify(data.entries);
+
+  localStorage.setItem('forms-entries', jsonEntries);
```

Comment:
You do not need to update `localStorage` after every entry is submitted, you should just do it right before the page unloads using a `beforeunload` event listener. Additionally, all `localStorage` actions should be done in the `data.js` file so that your data model is in the same file as your code for `localStorage`.

Also, you need to save the entire data model into localStorage, not just the `entries` property